src/components/checkbox/daikin-checkbox.ts

```diff
@@ -0,0 +1,198 @@
+import ctl from '@netlify/classnames-template-literals';
+import { LitElement, html, css, unsafeCSS } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+import tailwindStyles from '../../tailwind.css';
+import { cva, type VariantProps } from "class-variance-authority";
+import type { OmitNull } from "../../typeUtils";
+
+const baseCheckboxCN = ctl(`
+    appearance-none
+
+    inline-block
+    relative
+    rounded-sm
+    border-solid
+    border-2
+    border-daikinNeutral-400
+
+    hover:border-daikinBlue-300
+
+    active:border-daikinBlue-600
+    active:bg-daikinBlue-600
+
+    indeterminate:border-daikinBlue-600
+    indeterminate:bg-daikinBlue-600
+
+    checked:border-daikinBlue-600
+    checked:bg-daikinBlue-600
+
+    after:absolute
+    after:!w-full
+    after:!h-full
+    after:text-white
+
+    checked:after:i-daikin-checkbox-checked
+    indeterminate:after:i-daikin-checkbox-indeterminate
+
+    focus-visible:border-daikinBlue-700
+    focus-visible:outline-none
+
+    checked:focus-visible:border-daikinBlue-700
+    checked:focus-visible:bg-daikinBlue-700
+
+    checked:hover:border-daikinBlue-300
+    checked:hover:bg-daikinBlue-300
+    checked:active:border-daikinBlue-600
+    checked:active:bg-daikinBlue-600
+
+    indeterminate:active:border-daikinBlue-600
+    indeterminate:active:bg-daikinBlue-600
+    indeterminate:hover:border-daikinBlue-300
+    indeterminate:hover:bg-daikinBlue-300
+
+    indeterminate:focus-visible:border-daikinBlue-700
+    indeterminate:focus-visible:bg-daikinBlue-700
+
+    disabled:!border-daikinNeutral-200
+    disabled:!bg-white
+    indeterminate:disabled:!bg-daikinNeutral-200
+    checked:disabled:!bg-daikinNeutral-200
+`)
+
+const labelCN = cva(["leading-8", "not-italic", "font-normal", "align-middle"], {
+    variants: {
+            size: {
+                small: ["text-sm"],
+                large: ["text-base"],
+            },
+        },
+        defaultVariants: {
+            size: "small"
+        }
+});
+
+const checkboxCN = cva(baseCheckboxCN, {
+    variants: {
+            size: {
+                small: ["w-[18px]", "h-[18px]"],
+                large: ["w-5", "h-5"],
+            },
+        },
+        defaultVariants: {
+            size: "small"
+        }
+});
+
+type labelProps = OmitNull<VariantProps<typeof labelCN>>;
+type checkboxProps = OmitNull<VariantProps<typeof checkboxCN>>;
+type componentSize = labelProps["size"] & checkboxProps["size"];
```

Comment:
Type names should start with Uppercase.

```suggestion
type LabelProps = OmitNull<VariantProps<typeof labelCN>>;
type CheckboxProps = OmitNull<VariantProps<typeof checkboxCN>>;
type ComponentSize = LabelProps["size"] & CheckboxProps["size"];
```